js/src/components/contact-information/phone-number-card/verify-phone-number-content.test.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom/extend-expect';
+
+/**
+ * Internal dependencies
+ */
+import VerifyPhoneNumberContent from './verify-phone-number-content';
+
+describe( 'VerifyPhoneNumberContent component', () => {
+	test( 'typing Enter submits enclosing form', async () => {
+		const onVerificationStateChange = jest.fn();
+
+		render(
+			<VerifyPhoneNumberContent
+				verificationMethod="SMS"
+				onVerificationStateChange={ onVerificationStateChange }
+			/>
+		);
```

Comment:
`VerifyPhoneNumberContent` will issue the `requestPhoneVerificationCode` action when rendering, even though it fails on "TypeError: window.fetch is not a function" error and wouldn't really send the request to Google's production API, we could still consider mocking the action.

https://github.com/woocommerce/google-listings-and-ads/blob/e06126dc1f62b7fb98e98e03228d1b849c881d8c/js/src/components/contact-information/phone-number-card/verify-phone-number-content.js#L155-L160

Reply:
Why exactly we need to call  `handleVerificationCodeRequest` on render?

Reply:
Ignore my previous message. I was messing with another thing.

Reply:
I was trying to mockup `handleSubmit`, but it was unsuccessful. 

I tried with `VerifyPhoneNumberContent.handleSubmit = jest.fn()' but the component was still calling the previous `handleSubmit`.

After spending pretty much time investigating how to do so, I could not find any solution that doesn't require promoting `handleSubmit` as a component prop callback.  

Finally, I went back into the original @tomalec solution. (Adding the form and the submit button, then mockup the onsubmit)

Please let me know if there is a way to mockup `handleSubmit` function.

Reply:
> Please let me know if there is a way to mockup `handleVerifySubmit` function.

It could be mocked via the source actions of `'.~/data/actions'` or the exporting `{ useAppDispatch} from '.~/data'` which is composed as a **wp-data** store instance.

```js
// Method 1
jest.mock( '.~/data/actions', () => {
	return {
		requestPhoneVerificationCode: () => ( { type: 'mock' } ),
		verifyPhoneNumber: () => ( { type: 'mock' } ),
	};
} );

// Method 2
jest.mock( '.~/data', () => {
	return {
		useAppDispatch: () => {
			return {
				async requestPhoneVerificationCode() {},
				async verifyPhoneNumber() {},
			};
		},
	};
} );
```

Reply:
Nice, TYVM @eason9487 

I will use this when creating `verify-phone-number-content.js` tests.